examples/advanced/example_model_optimization.py

```diff
@@ -305,28 +315,20 @@ def train(
 #
 # .. note::
 #
-#   If you want to perform the training yourself, set
-#   ``use_pretrained_transformer=False``. If you do, you also need to replace
-#   ``dataset = None`` below with the dataset you want to train on.
+#   If you want to perform the training yourself, set ``root`` to the location
+#   of the images downloaded from the link below
 #
-# .. note::
-#
-#   The weights of the provided transformer were trained with the
-#   `2014 training images <http://images.cocodataset.org/zips/train2014.zip>`_ of the
-#   `COCO dataset <https://cocodataset.org/>`_. The training was performed for
-#   ``num_epochs=2`` and ``batch_size=4``. Each image was center-cropped to
-#   ``256 x 256`` pixels.
 
-use_pretrained_transformer = True
+root = None
 checkpoint = "example_transformer.pth"
 
-if use_pretrained_transformer:
+
+def load(checkpoint):
     if path.exists(checkpoint):
-        state_dict = torch.load(checkpoint)
-    else:
-        # Unfortunately, torch.hub.load_state_dict_from_url has no option to disable
-        # printing the downloading process. Since this would clutter the output, we
-        # suppress it completely.
+        return torch.load(checkpoint)
+
+    if checkpoint == "example_transformer.pth":
```

Comment:
What is the use case for this extra check?

Reply:
Its about how best to handle having to download the example transformer. 
If we reach this point we know that the specified transformer does not exist on disk and is the example, so it'll grab it.
We could make this functionality more explicit with a separate download function and a note that this is how to get it?

Reply:
> If we reach this point we know that the specified transformer does not exist on disk and is the example, so it'll grab it.

Maybe I should have been more explicit. Why are we only doing that when the checkpoint matches the default name? Why not always download in case the checkpoint is not available?

> We could make this functionality more explicit with a separate download function and a note that this is how to get it?

Makes sense. Go for it!